### DISCUSSION TITLE
Get verbose log output when running backup through the API

### DISCUSSION THREAD

**Discussion by bn-gmoss** [Altinity/clickhouse-backup#504] [Q&A] [answered]
Hello,

I'd like to use this tool to manage backup and restore of several different clickhouse shards my team currently runs on a kubernetes cluster. To streamline and simplify things, we've decided to use the API, but we'd like to get the same verbose output we get when running the cli on a shard pod directly. 

When run on a specific shard, the cli outputs a bunch of great log info detailing the steps of the backup process along with timestamps for each action.

Is there a way to capture this information when working through the API? At most, the /backup/status endpoint will return a fairly simple json object containing the backup name, when it started, and when it ended, but we don't get insight into the steps taken during the backup.

Is there a flag we can trip to get this verbose output?

If not, this would be a great quality of life feature to add, as collecting verbose debug data will help us ensure we haven't taken an incomplete backup and help identify what happened when something goes wrong.

Thank you

**Comment by Slach** [comment, marked as answer]
use in your pod template

```yaml
spec:
  containers:
    - name: clickhouse-backup
      image: altinity/clickhouse-backup:latest
      environment:
        - LOG_LEVEL=debug
```

or configure via ConfigMap /etc/clickhouse-backup/config.yml
```
general:
  log_level: debug
```  

`api server` show their log to stdout of container

**Reply by bn-gmoss** [reply]
Yeah we have this configured currently, but i was hoping there was a way to have the api server return this log level data along with the api return json to the entity that sent the request.

**Reply by Slach** [reply]
Upload and download operations are asynchronous, we can't catch logs and return it to API, logs could be a massive
So, just use standard `kubernetes` approach and collect pods log with `fluentd + elasticsearch` or with `vector + clickhouse` visualize it in `grafana`

**Reply by bn-gmoss** [reply]
that makes sense, we've been just trying this out on small test clusters so far but i see what you mean about logs being too big to shift easily. 

thanks for your recommendations and help
